### PR TITLE
[Arm64] Implement Poison()

### LINKED ISSUE #13117: [ARM64/Linux] Implement UMEntryThunkCode::Poison
#12731 introduces ``UMEntryThunkCode::Poison`` for all the architecture (x86, x64, arm, arm64), but arm64 variant does not have a proper implementation.

We need to implement it properly.

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes #13117

### CODE REVIEW

File: src/vm/arm64/stubs.cpp [hunk 1]
```diff
@@ -1246,7 +1246,8 @@ void UMEntryThunkCode::Encode(BYTE* pTargetCode, void* pvSecretParam)
 
 void UMEntryThunkCode::Poison()
 {
-
+    // Insert 'brk 0xbe' at the entry point
+    m_code[0] = 0xd42017c0;
```
Comment (sdmaclea) [inline src/vm/arm64/stubs.cpp]: @parjong It is not clear to me why none of the `Poison()` functions call `FlushInstructionCache()`.  I assume you know it is not necessary.

Comment (jkotas) [inline src/vm/arm64/stubs.cpp]: Discussed here: https://github.com/dotnet/coreclr/pull/12731#discussion_r128414860

**Comment by sdmaclea** [comment]
@dotnet-bot test Tizen armel Cross Release Build

**Comment by sdmaclea** [comment]
@dotnet-bot test Windows_NT x64 Formatting

File: src/vm/arm64/stubs.cpp (hunk 1 above)
Comment (sdmaclea) [inline src/vm/arm64/stubs.cpp]: Acknowledged

**Pull request merged**